src/care/host_device_map.h

```diff
@@ -136,7 +143,11 @@ namespace care {
 
         // emplace a key value pair, using return of atomic increment to provide the initial insertion index
         inline CARE_DEVICE void emplace(key_type key, mapped_type val) const {
-           int index = ATOMIC_ADD(m_size_ptr[0], 1);
+           care::local_ptr<int> size_ptr = m_size_ptr;
+           int index = ATOMIC_ADD(size_ptr[0], 1);
+           if (size_ptr[0] > m_max_size) {
```

Comment:
Same here.

Reply:
But I think we would definitely want this active in our default builds.

Reply:
Oof, to have it active for debug builds but turn it off for release builds is surprisingly difficult (especially with explicit instantiations and public packages).

Reply:
I'm just worried that since emplace is often called in a loop, the overhead of bounds checking will be significant.